<!-- Core launch file to run MSL-RAPTOR node -->
<launch>
  <arg name="id"                   />
  <arg name="robot_type"           />
  <arg name="ns"                  value="/$(arg robot_type)$(arg id)" />
  <arg name="classes_names_file"  default="$(find msl_raptor)/params/objects/classes_coco_mslquad.names" /> <!-- yaml file containing all the object sizes -->
  <arg name="object_sizes_file"   default="$(find msl_raptor)/params/objects/object_params.yaml" /> <!-- yaml file containing all the object sizes -->
  <arg name="object_used_filepath" default="$(find msl_raptor)/params/objects/objects_used/" /> <!-- yaml file containing the names of objects used -->
  <arg name="object_used_file"    default="objects_used.txt" /> <!-- yaml file containing the names of objects used -->
  <arg name="experiment_param_file"    default="experiment.yaml" /> <!-- yaml file containing the names of objects used -->
  <arg name="b_record"            default="false" />  <!-- boolean if we are recording data in rosbag -->
  <arg name="bag_out_folder"         default="/mounted_folder/raptor_processed_bags" />  <!-- path to the rosbag -->

  <!-- Launch our main node -->
  <node name="MSL_RAPTOR" pkg="msl_raptor" type="msl_raptor_main.py" output="screen">
    <param name="ns"  value="$(arg ns)"/> 
    <rosparam file="$(find msl_raptor)/params/$(arg experiment_param_file)" />
    <param name="object_sizes_file"  value="$(arg object_sizes_file)"/>
    <param name="object_used_file"  value="$(arg object_used_filepath)$(arg object_used_file)"/>
    <param name="classes_names_file"  value="$(arg classes_names_file)"/>
    <rosparam command="load" file="$(find msl_raptor)/params/setup/$(arg robot_type)$(arg id).yaml" />
  </node>

    <!-- IF we should record a rosbag of results -->
  <group if="$(arg b_record)">
    <group if="$(arg b_record)">
      <!-- if FROM a rosbag (i.e. post-processed), then use the name of the source rosbag to name the newly output bag -->
      <node pkg="rosbag" type="record" name="msl_raptor_recording" args="-O $(arg bag_out_folder)/msl_raptor_output_ -a"/>
    </group>
  </group>
</launch>